src/Arcus.EventGrid.Publishing/EventGridPublisher.cs

```diff
@@ -89,14 +89,35 @@ public async Task PublishRawAsync(string eventId, string eventType, string event
 
             var rawEvent = new RawEvent(eventId, eventType, eventBody, eventSubject, dataVersion, eventTime);
 
-            IEnumerable<RawEvent> eventList = new List<RawEvent>
-            {
-                rawEvent
-            };
+            return PublishRawAsync(rawEvent);
```

Comment:
Just wondering:
Why is this method not declared as `async` and are you not `awaiting` the `PublishRawAsync` method ?
I know it might not be necessary to do so, since you're already returning an awaited Task, but I just wonder ....

Reply:
A small improvement performance wise that you see sometimes with delegated asynchronous methods. Since, like you said, the client can start the awaiting and some internal method that only return this.

Reply:
This explains the difference: https://blog.stephencleary.com/2016/12/eliding-async-await.html

His recommendation is the following:
> Do not elide by default. Use the async and await for natural, easy-to-read code.

I'm not an expert on the topic but tend to agree, certainly given exceptions are not thrown on the await but when the method returns which can give unexpected behavior as a consumer

Reply:
Can you add the async/await please @stijnmoreels 	?

Reply:
? I thought we agreed on the fact that it is indeed better to not use async/await on overloads? Or is this again something people will question every time they see it?

Reply:
I don't recall on discussing this before to be honest, but maybe I just forgot about it.

However, based on the post mentioned above they recommend not to elide async/await by default

Reply:
Let's be pragmatic, how is it done for other methods today?

Reply:
It was just a question that I've raised :)  To be honest, personally, I would -by default- await on async code, even in overloaded methods.  If I read the article of Stephen Cleary mentionned by Tom, I'm even more convinced to do that.

However, I've been looking into the source-code of [`HttpClient`](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/HttpClient.cs) and [`QueueClient`](https://github.com/Azure/azure-service-bus-dotnet/blob/dev/src/Microsoft.Azure.ServiceBus/QueueClient.cs) just to have an idea what's being done there, and there, the async/await is omitted in overloaded methods.

In fact, if everybody is OK with omitting it, it's fine for me as well.  I don't feel we should block this PR just for this. :)

Reply:
Any update on this @stijnmoreels? Are we eliding for other methods or not?

Reply:
Based on that, we should just align and create an issue to no longer elide given what the post mentions, for Arcus the performance hit will never be an issue.

If it is, we can still tweak then but we should avoid it given it gives unexpected results for consumers.

Reply:
It's all the same for me. It becomes a bit of a habit for me to remove the async/await part of overloaded methods. But I can certainly live with adding them.

Reply:
Let's add them so we can merge this PR then!

Reply:
Thank you!